### PR TITLE
add php benchmark

### PR DESCRIPTION
**Pull request by BSBandme**
Benchmark results with c extension:
```
PHP protobuf benchmark result:

result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/datasets/google_message2/dataset.google_message2.pb
   Throughput for benchmark parse_php: 111.84 MB/s
   Throughput for benchmark serailize_php: 210.17 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/datasets/google_message1/proto3/dataset.google_message1_proto3.pb
   Throughput for benchmark parse_php: 14.41 MB/s
   Throughput for benchmark serailize_php: 21.91 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/datasets/google_message1/proto2/dataset.google_message1_proto2.pb
   Throughput for benchmark parse_php: 14.84 MB/s
   Throughput for benchmark serailize_php: 22.52 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message4.pb
   Throughput for benchmark parse_php: 117.44 MB/s
   Throughput for benchmark serailize_php: 180.51 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message3_4.pb
   Throughput for benchmark parse_php: 1903.16 MB/s
   Throughput for benchmark serailize_php: 3091.22 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message3_3.pb
   Throughput for benchmark parse_php: 11.89 MB/s
   Throughput for benchmark serailize_php: 30.06 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message3_5.pb
   Throughput for benchmark parse_php: 3.43 MB/s
   Throughput for benchmark serailize_php: 11.91 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message3_2.pb
   Throughput for benchmark parse_php: 75.69 MB/s
   Throughput for benchmark serailize_php: 87.16 MB/s
result for test data file: /usr/local/google/home/yilunchong/mygit/protobuf/benchmarks/tmp/proto3_data/dataset.google_message3_1.pb
   Throughput for benchmark parse_php: 108.19 MB/s
   Throughput for benchmark serailize_php: 108.04 MB/s
```

### CODE REVIEW

File: benchmarks/php/PhpBenchmark.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            substr($path, -strlen(NAME)) != NAME) {
+            include(substr($path, strlen($prefix) + 1));
+        } elseif (is_dir($path)) {
+            _require_all($path, $prefix);
+        }
+    }
+}
+// include all file 
+foreach (explode(PATH_SEPARATOR, get_include_path()) as $one_include_path) {
+    _require_all($one_include_path, $one_include_path);
```
Comment (BSBandme) [inline benchmarks/php/PhpBenchmark.php]: Not sure whether this is the correct including way.

**Comment by BSBandme** [comment]
How can I add php benchmark with c extension?

**Comment by TeBoring** [comment]
@BSBandme just go to php/ext/google/protobuf, then do:
```
phpize && ./configure CFLAGS='-O3' && make
```
Then you will have a protobuf.so in modules folder.
In command line, when you run php, add:
```
-dextension=<path to protobuf.so>
```

**Comment by TeBoring** [comment]
What's the results for cpp?

**Comment by TeBoring** [comment]
How do you include php runtime now?

**Comment by BSBandme** [comment]
Sorry that the previous result seems already linked c extension. 
I just `sudo pecl install protobuf-3.6.0` to install php protobuf.

So how can I test without c extension?

The cpp result can be found [here](https://github.com/google/protobuf/blob/master/docs/performance.md) (test under the same machine).

**Comment by TeBoring** [comment]
currently, php doesn't support proto2. How do you get result for proto2 message?

**Comment by TeBoring** [comment]
Check here for how to run without c extension: https://github.com/google/protobuf/blob/master/conformance/Makefile.am#L346
Note that you need to uninstall c extesnion

**Comment by BSBandme** [comment]
Done autoload fix

**Comment by TeBoring** [comment]
Are these php tests still failing?

**Comment by BSBandme** [comment]
Tests fixed.

**Pull request merged**